timeout.go

```diff
@@ -26,31 +22,33 @@ type TimerWheel struct {
 	wheelDuration time.Duration
 
 	// The actual wheel which is just a set of singly linked lists, head/tail pointers
-	wheel []*TimeoutList
+	wheel []*TimeoutList[T]
 
 	// Singly linked list of items that have timed out of the wheel
-	expired *TimeoutList
+	expired *TimeoutList[T]
 
 	// Item cache to avoid garbage collect
-	itemCache   *TimeoutItem
+	itemCache   *TimeoutItem[T]
 	itemsCached int
 }
 
 // TimeoutList Represents a tick in the wheel
-type TimeoutList struct {
-	Head *TimeoutItem
-	Tail *TimeoutItem
+type TimeoutList[T any] struct {
+	Head *TimeoutItem[T]
+	Tail *TimeoutItem[T]
 }
 
 // TimeoutItem Represents an item within a tick
-type TimeoutItem struct {
-	Packet firewall.Packet
-	Next   *TimeoutItem
+type TimeoutItem[T any] struct {
+	Item T
```

Comment:
Did you consider making `Item` type `any`?
```
type TimeoutList struct {
        Head *TimeoutItem
        Tail *TimeoutItem
}

type TimeoutItem struct {
        Item any
        Next *timeoutItem
}
```

Reply:
Not threading `T` through to it requires a type assertion at `TimerWheel.Purge` and incurs a small performance penalty.

```
name                 old time/op  new time/op  delta
TimerWheel_Purge-10  2.49ns ± 0%  2.63ns ± 0%  +5.69%  (p=0.000 n=19+17)
```

There's also a bit of awkwardness since `TimerWheel.Add` currently returns the added `TimeoutItem`. `TimeoutItem.Item` could be changed to something that can't be casted to `T`. Nothing uses the returned `TimeoutItem` today so we could remove that.